src/Php74/Php74.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Php74;
+
+/**
+ * @author Ion Bazan <ion.bazan@gmail.com>
+ *
+ * @internal
+ */
+final class Php74
+{
+    /**
+     * @param object $obj
+     *
+     * @return array
+     */
+    public static function get_mangled_object_vars($obj)
+    {
+        if (!$obj instanceof \ArrayIterator && !$obj instanceof \ArrayObject) {
```

Comment:
I would rather write the special cases here, and have the final code path being the generic one. This will avoid having to update this condition each time we need to add another special case.

Reply:
I flipped the condition. If a new special case appears, we should consider moving it to a new method.